src/delta.rs

```diff
@@ -448,11 +448,19 @@ fn handle_hunk_header_line(
     } else if config.hunk_header_style.is_omitted {
         writeln!(painter.writer)?;
     } else {
-        let line = match painter.prepare(&raw_code_fragment, false) {
-            s if !s.is_empty() => format!("{} ", s),
-            s => s,
+        // Keep the raw_line structure when color_only mode
+        let line = if config.color_only {
+            format!(" {}", &line)
+        } else {
+            match painter.prepare(&raw_code_fragment, false) {
+                s if !s.is_empty() => format!("{} ", s),
+                s => s,
+            }
```

Comment:
Question about code before your change: do you know why `painter.prepare()` is being called here? (Is it to expand tabs?).

Reply:
For documentation purposes, can you explain the reason that we don't call `painter.prepare()` under `color-only`?

Reply:
@dandavison 
The reason is this (it may be long, sorry).

#### ① `can you explain the reason that we don't call painter.prepare() under color-only?`

Simply it's because it doesn't need to use `painter.prepare()`. Because we won't change structure from raw_line, and will not use decoration.

#### ② then `do you know why painter.prepare() is being called here? (Is it to expand tabs?)`

I think it exists for expanding tabs `just in the case there are tabs` (just my speculation, I couldn't find exact purpose in `git blame`). However, this doesn't need it because hunk header will never have tabs even when it's nested.

So, it actually could be written like below.
```rust
match raw_code_fragment {
    s if !s.is_empty() => format!("{} ", s),
    s => s,
}

// This change passes all tests in fact.
```

This match condition is necessary, because some options will be broken such as box.
<img width="626" alt="ss" src="https://user-images.githubusercontent.com/41639488/101948966-4183f680-3c36-11eb-9ece-8ad6be6b3478.png">

Reply:
> So, it actually could be written like below.
 > // This change passes all tests in fact.

Great. Let's bear that in mind for a little future cleanup. Then we can get rid of that second argument to `prepare`.